samples/matter/lock/CMakeLists.txt

```diff
@@ -68,4 +68,27 @@ chip_configure_data_model(app
     INCLUDE_SERVER
     ZAP_FILE ${CMAKE_CURRENT_SOURCE_DIR}/src/lock.zap
 )
+
+if(CONFIG_THREAD_WIFI_SWITCHING)
+    if(CONFIG_NET_L2_OPENTHREAD)
+        set(app_no 1)
+    else()
```

Comment:
How about using `elseif(CONFIG_CHIP_WIFI)` and else() -> some error message?

Reply:
Hmm. I think I'll rather add another hidden Kconfig to encode the application number. You also added similar conditions in the source code, so maybe it's better to move this into a single place - Kconfig.

Reply:
Good idea, we will avoid duplicating the same check in several places.

Reply:
Added hidden Kconfig options and reduce some ifdefs in the code.